### PR TITLE
Fix element wrapping section headers

### LINKED ISSUE #958: Fix elements wrapping in section headers
image

From the picture above we can see that the expand button is moved to the next line. 
This happens when the navigator width is small.

We should probably strip characters from the section name and add ellipsis.

### PR DESCRIPTION
**Pull request by andreiv1992**
This PR...

## Changes

Added ellipsis to section headers

## Fixes

#958 
image 

## How to test it

Make the window really small

## Screenshots

image 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by olensmar** [comment]
thanks @andreiv1992 - it looks like the cluster label has moved up to the right from below the "K8s Resources" label

image

**Comment by andreiv1992** [comment]
Thanks a lot for catching that Ole.

Found a solution and it is working but I think it would be best if we can create a separate property(eg HeaderContext), similar to NameSuffix where we can display information like this on a new line. It is cumbersome to display information on a new line when it is in the middle of other inline components.

Let me know what you think

**Comment by olensmar** [comment]
thanks @andreiv1992 - that sounds good to me - do you want to do that in this PR or start a separate branch after merging? @devcatalin what do you think?

File: src/components/molecules/SectionRenderer/SectionHeader.tsx [hunk 1]
```diff
@@ -65,6 +65,15 @@ function SectionHeader(props: SectionHeaderProps) {
       dispatch(sectionInstance.checkable.uncheckItemsAction);
     }
   }, [sectionInstance, dispatch]);
+  let shouldDisplayPlus = false;
```
Comment (devcatalin) [inline src/components/molecules/SectionRenderer/SectionHeader.tsx:68]: could you convert these to `useMemo` ?

**Comment by devcatalin** [comment]
> thanks @andreiv1992 - that sounds good to me - do you want to do that in this PR or start a separate branch after merging? @devcatalin what do you think?

sounds good to me, as long as we don't break the UI somewhere

File: src/components/molecules/SectionRenderer/SectionHeader.tsx (hunk 1 above)
Comment (andreiv1992) [reply src/components/molecules/SectionRenderer/SectionHeader.tsx:68]: Sure

**Comment by andreiv1992** [comment]
> 

I'd like to do it in a different PR

**Review by olensmar — APPROVED** [review]

**Pull request merged**